idx/memory/memory.go

```diff
@@ -864,18 +864,19 @@ func (m *UnpartitionedMemoryIdx) add(archive *interning.ArchiveInterned) {
 	return
 }
 
-// Get returns an interning.Archive that matches the supplied schema.MKey.
-// Upon failure it returns an empty interning.Archive
-func (m *UnpartitionedMemoryIdx) Get(id schema.MKey) (interning.Archive, bool) {
+// Get returns an *interning.ArchiveInterned that matches the supplied schema.MKey.
+// Upon failure it returns nil and false
+func (m *UnpartitionedMemoryIdx) Get(id schema.MKey) (*interning.ArchiveInterned, bool) {
 	pre := time.Now()
 	m.RLock()
-	defer m.RUnlock()
 	def, ok := m.defById[id]
 	statGetDuration.Value(time.Since(pre))
 	if ok {
-		return def.GetArchive(), ok
+		m.RUnlock()
+		return def, ok
```

Comment:
doesn't that still need to be `return def.CloneInterned(), ok`?

Reply:
Yes, maybe. If we do that I would like to create another method that just returns SchemaId, AggId, and Interval then. Otherwise we will be facing the same Clone / Release problem as this is trying to solve.